src/Client/Updater.php

```diff
@@ -129,14 +119,99 @@ public function validateTarget($targetRepoPath, $targetStream)
         $timestampStructure = json_decode($timestampContents, true);
         // SPEC: 2.1
         if (! $this->checkSignatures($timestampStructure, 'timestamp')) {
-          // Exception? Log + return false?
             throw new \Exception("Improperly signed repository timestamp.");
         }
 
+        // SPEC: 2.2
+        $currentStateTimestamp = json_decode($this->durableStorage['timestamp.json'], true);
+        $this->checkRollbackAttack($currentStateTimestamp['signed'], $timestampStructure['signed']);
+
+        // SPEC: 2.3
+        $this->checkFreezeAttack($timestampStructure['signed'], $nowDate);
+        $durableStorage['timestamp.json'] = $timestampContents;
```

Comment:
`checkSignatures()` returns a bool and then `refresh()` throws an exception if it's wrong. However, the new checks return `void` and throw the exceptions themselves. We should make the behavior consistent. Thoughts on which is best? 

Returning booleans might give us more flexibility for testing and reuse, but OTOH we can always `try/catch`. Always throwing the exceptions within the checkers might be safer, because it ensures a caller can't accidentally introduce a bypass of the check. What do you think?

Reply:
I like the idea of throwing exceptions

Reply:
We're thowing exceptions in any case.. it's a question as to where in the callstacks the exception is thrown: within the checker, or in response to the checker's result. I assming Ted means "Exceptions thrown  by the checker,  not the caller."

Reply:
@tedbow Can you clarify what you meant?

Reply:
I meant inside the method not the caller.

Reply:
Changed \Tuf\Client\Updater::checkSignatures to throw an exceptions like the other check methods